packages/flutter/test/cupertino/segmented_control_test.dart

```diff
@@ -894,6 +906,357 @@ void main() {
     semantics.dispose();
   });
 
+  testWidgets('Non-centered taps work on smaller widgets', (WidgetTester tester) async {
+    final Map<int, Widget> children = <int, Widget>{};
+    children[0] = const Text('A');
+    children[1] = const Text('B');
+
+    int sharedValue = 1;
+
+    await tester.pumpWidget(
+      new StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return boilerplate(
+            child: new SegmentedControl<int>(
+              key: const ValueKey<String>('Segmented Control'),
+              children: children,
+              onValueChanged: (int newValue) {
+                setState(() {
+                  sharedValue = newValue;
+                });
+              },
+              groupValue: sharedValue,
+            ),
+          );
+        },
+      ),
+    );
+
+    expect(sharedValue, 1);
+
+    final double childWidth = getRenderSegmentedControl(tester).firstChild.size.width;
+    final Offset centerOfSegmentedControl = tester.getCenter(find.text('A'));
+
+    // Tap just inside segment bounds
+    await tester.tapAt(new Offset(childWidth - 10.0, centerOfSegmentedControl.dy));
+
+    expect(sharedValue, 0);
+  });
+
+  testWidgets('Animation is correct when the selected segment changes',
+          (WidgetTester tester) async {
+    await tester.pumpWidget(setupSimpleSegmentedControl());
+
+    await tester.pumpAndSettle();
+
+    await tester.tap(find.text('Child 2'));
+
+    await tester.pump();
+    expect(getBackgroundColor(tester, 0), CupertinoColors.activeBlue);
+    expect(getBackgroundColor(tester, 1), const Color(0x33007aff));
+
+    await tester.pump(const Duration(milliseconds: 40));
+    expect(getBackgroundColor(tester, 0), const Color(0xff3d9aff));
+    expect(getBackgroundColor(tester, 1), const Color(0x64007aff));
+
+    await tester.pump(const Duration(milliseconds: 40));
+    expect(getBackgroundColor(tester, 0), const Color(0xff7bbaff));
+    expect(getBackgroundColor(tester, 1), const Color(0x95007aff));
+
+    await tester.pump(const Duration(milliseconds: 40));
+    expect(getBackgroundColor(tester, 0), const Color(0xffb9daff));
+    expect(getBackgroundColor(tester, 1), const Color(0xc7007aff));
+
+    await tester.pump(const Duration(milliseconds: 40));
+    expect(getBackgroundColor(tester, 0), const Color(0xfff7faff));
+    expect(getBackgroundColor(tester, 1), const Color(0xf8007aff));
+
+    await tester.pump(const Duration(milliseconds: 40));
+    expect(getBackgroundColor(tester, 0), CupertinoColors.white);
+    expect(getBackgroundColor(tester, 1), CupertinoColors.activeBlue);
+  });
```

Comment:
(see "stall" comment above): Consider a variant of this test where instead of calling `pump` each time, you call `pumpWidget` (calling `setupSimpleSegmentedControl` each time and passing the same durations as now). The two tests should have exactly the same expectations.

Reply:
Okay yes, so that causes an issue because initState() runs each time pumpWidget is called. I'm a bit at a loss for how to fix this though - is there a way you know of to get the animations to run correctly?